Datastore

GuillermoTafoya opened the pull request:
Defined the classes for the database, defined methods for retrieving, deleting and updating users; and defined the connection for a login/signup form.

Register Page
![image](https://user-images.githubusercontent.com/63993813/176987399-60e7c122-5573-4707-a317-0310165297b1.png)
User Info Page (For Debugging Purposes)
![image](https://user-images.githubusercontent.com/63993813/176987429-967d2288-5872-46af-895a-28ba58224e3d.png)

GuillermoTafoya committed:
Defined the classes for the database, defined methods for retrieving, deleting and updating users; and defined the connection for a login/signup form.

cindyxue commented on src/main/java/com/google/sps/servlets/DeleteAll.java:
```diff
@@ -0,0 +1,32 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.DatastoreService;
+//import com.google.sps.data.Day;
```
Remove the comments?

GuillermoTafoya replied:
Yup, thanks. Removed innecesary commented code! :)

cindyxue commented on src/main/java/com/google/sps/data/DatastoreService.java:
```diff
@@ -0,0 +1,208 @@
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+//import com.google.cloud.datastore.StructuredQuery.OrderBy;
```
nit: remove the comment?

cindyxue commented on src/main/java/com/google/sps/data/DatastoreService.java:
```diff
@@ -0,0 +1,208 @@
+            .build();
+        QueryResults<Entity> results = datastore.run(query);
+        if (results.hasNext()) {
+            Entity entity = results.next();
+
+            String email = entity.getString("email");
+            String passwordHash = entity.getString("passwordHash");
+
+            String firstName = entity.getString("firstName");
+            String middleName = entity.getString("middleName");
+            String lastName = entity.getString("lastName");
+
+            var calendar = entity.getList("calendar");
+            var journal = entity.getList("journal");
+
+            long panicButtonPressed = entity.getLong("panicButtonPressed");
+
+            User user = new User(username, email, passwordHash, firstName, middleName, lastName, calendar, journal, panicButtonPressed);
```
Refactor this into a method (i.e. createUser)? You can replace line 28-41 with this method too to reduce duplicate code.

GuillermoTafoya replied:
Done! :D 
Wrote two methods to keep the code a bit cleaner, thanks!

cindyxue commented on src/main/java/com/google/sps/data/DatastoreService.java:
```diff
@@ -0,0 +1,208 @@
+                .set("journal", user.getJournal())
+                .set("panicButtonPressed", user.getPanicButtonPressed())
+                .build();
+        // datastore.add(user); // Datastore add operation: inserts the provided entities.
```
nit: remove this?

cindyxue reviewed (commented):
Thanks for working on this! Overall looks good to me!

Could you please add a screenshot of the registration page in the PR description?

GuillermoTafoya committed:
.Defined the classes for the database, defined methods for retrieving, deleting and updating users; and defined the connection for a login/signup form. Cleaned code according to the PR review.

GuillermoTafoya committed:
.Defined the classes for the database, defined methods for retrieving, deleting and updating users; and defined the connection for a login/signup form. Cleaned code according to the PR review.

GuillermoTafoya committed:
Added Handlers for Login and Register

Added handlers for login and register, started files to manage the calendar and journal.

GuillermoTafoya committed:
Handle payload for login and register servlets.

GuillermoTafoya committed:
Login and register forms working correctly. Redirects to /debugger.

GuillermoTafoya committed:
Merge pull request #2 from cindyxue/guillermo-datastore-login-register

Login and Register pages working correctly. Redirects to /debugger afterwards.

GuillermoTafoya committed:
Added a base class to inherit methods that allow payload loading and JSON return.

GuillermoTafoya committed:
Merge pull request #3 from cindyxue/guillermo/datastore/servletTemplate-inheritance

Servlet Template Base Class:  Added a base class to inherit methods that allow payload loading and JSON return.

GuillermoTafoya committed:
Added cookie management for user data. Created a js module to make API calls easier. Created a debugging page to demonstrate all API features. Handle user updates.

GuillermoTafoya committed:
Added functions to delete user and delete cookies(user logout).

GuillermoTafoya committed:
Comments added for Datastore-API.js

GuillermoTafoya committed:
Comments added for Datastore-API.js

GuillermoTafoya committed:
Added api documentation on /api/documentation and /api/documentation.pdf
